WordPress/WordPressTest/ViewRelated/Post/Utils/PostNoticeViewModelTests.swift

```diff
@@ -114,6 +114,30 @@ class PostNoticeViewModelTests: XCTestCase {
         expect(postCoordinator.cancelAutoUploadOfInvocations).to(equal(1))
     }
 
+    func testFailedPublishedUploadedDraftPostsPublishButtonWillMarkForAutoUpload() {
+        // Given
+        let context = ContextManager.shared.mainContext
+        let post = PostBuilder(context)
+            .drafted()
+            .with(title: "I've been drafted!")
+            .withRemote()
+            .with(remoteStatus: .sync)
+            .with(userName: "Meee")
+            .with(password: "ThisIsABadPassword")
```

Comment:
I was wondering if we really needed username/password in the test. I tried removing it and got a `caught "NSInternalInconsistencyException", "Invalid parameter not satisfying: username != nil"`.

But we can fix that by mocking the `save` method in `MockPostCoordinator`:

````
private final class MockPostCoordinator: PostCoordinator {
        (...)

        override func save(post postToSave: AbstractPost) {

        }
    }
```

Which I think is the behavior we expect (we're really not interested in the save logics).

This will even make these tests faster. :)